hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -674,6 +685,624 @@ public Response getDeletedDirectorySummary() {
+      long replicatedTotal = 0;
+      long unreplicatedTotal = 0;
+      boolean keysFound = false; // Flag to track if any keys are found
+
+      // Search keys from non-FSO layout.
+      Map<String, OmKeyInfo> obsKeys;
+      Table<String, OmKeyInfo> keyTable =
+          omMetadataManager.getKeyTable(BucketLayout.LEGACY);
+      obsKeys = retrieveKeysFromTable(keyTable, paramInfo);
+      for (Map.Entry<String, OmKeyInfo> entry : obsKeys.entrySet()) {
+        keysFound = true;
+        KeyEntityInfo keyEntityInfo =
+            createKeyEntityInfoFromOmKeyInfo(entry.getKey(), entry.getValue());
+
+        listKeysResponse.getKeys().add(keyEntityInfo);
+        replicatedTotal += entry.getValue().getReplicatedSize();
+        unreplicatedTotal += entry.getValue().getDataSize();
+      }
+
+      // Search keys from FSO layout.
+      Map<String, OmKeyInfo> fsoKeys = searchKeysInFSO(paramInfo);
+      for (Map.Entry<String, OmKeyInfo> entry : fsoKeys.entrySet()) {
+        keysFound = true;
+        KeyEntityInfo keyEntityInfo =
+            createKeyEntityInfoFromOmKeyInfo(entry.getKey(), entry.getValue());
+
+        listKeysResponse.getKeys().add(keyEntityInfo);
+        replicatedTotal += entry.getValue().getReplicatedSize();
+        unreplicatedTotal += entry.getValue().getDataSize();
+      }
+
+      // If no keys were found, return a response indicating that no keys matched
+      if (!keysFound) {
+        return ReconResponseUtils.noMatchedKeysResponse(paramInfo.getStartPrefix());
+      }
+
+      // Set the aggregated totals in the response
+      listKeysResponse.setReplicatedDataSize(replicatedTotal);
+      listKeysResponse.setUnReplicatedDataSize(unreplicatedTotal);
+
+      return Response.ok(listKeysResponse).build();
+    } catch (IOException e) {
+      return ReconResponseUtils.createInternalServerErrorResponse(
+          "Error listing keys from OM DB: " + e.getMessage());
+    } catch (RuntimeException e) {
+      return ReconResponseUtils.createInternalServerErrorResponse(
+          "Unexpected runtime error while searching keys in OM DB: " + e.getMessage());
+    } catch (Exception e) {
+      return ReconResponseUtils.createInternalServerErrorResponse(
+          "Error listing keys from OM DB: " + e.getMessage());
+    }
+  }
+
+  public Map<String, OmKeyInfo> searchKeysInFSO(ParamInfo paramInfo)
+      throws IOException {
+    int originalLimit = paramInfo.getLimit();
+    Map<String, OmKeyInfo> matchedKeys = new LinkedHashMap<>();
+    // Convert the search prefix to an object path for FSO buckets
+    String startPrefixObjectPath = convertStartPrefixPathToObjectIdPath(paramInfo.getStartPrefix());
+    String[] names = parseRequestPath(startPrefixObjectPath);
+    Table<String, OmKeyInfo> fileTable =
+        omMetadataManager.getKeyTable(BucketLayout.FILE_SYSTEM_OPTIMIZED);
+
+    // If names.length > 2, then the search prefix is at the volume or bucket level hence
```

Comment:
the search prefix supported is at bucket level, plz update the comment

Reply:
ok done.